h2o-parsers/h2o-parquet-parser/src/main/java/water/parser/parquet/ChunkConverter.java

```diff
@@ -180,6 +186,15 @@ public void addBinary(Binary value) {
     public void addLong(long value) {
       _writer.addNumCol(_colIdx, value, 0);
     }
+
+    @Override
+    public void addBinary(Binary value) {
+      final NanoTime nanoTime = NanoTime.fromBinary(value);
+      final long millisecondsToMidnight = DateTimeUtils.fromJulianDay(nanoTime.getJulianDay() - JULIAN_CALENDAR_INTEGER_ROUNDUP);
+      long millisecondsAfterMidnight =nanoTime.getTimeOfDayNanos() / NANOS_IN_MILLISECOND;
+      
+      _writer.addNumCol(_colIdx, millisecondsToMidnight + millisecondsAfterMidnight);
```

Comment:
please also see the alternative implementation in Presto: https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetTimestampUtils.java

what they are doing will be faster, but this is definitely fine as well